locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/JcloudsLocation.java

```diff
@@ -1551,6 +1552,37 @@ public Template buildTemplate(ComputeService computeService, ConfigBag config) {
         return template;
     }
 
+    protected String toStringNice() {
+        Entities.dumpInfo(this);
+        String s = config().get(ORIGINAL_SPEC);
+        if (Strings.isBlank(s)) s = config().get(NAMED_SPEC_NAME);
+        if (Strings.isBlank(s)) s = config().get(FINAL_SPEC);
+        if (Strings.isBlank(s)) s = getDisplayName();
+        
+        String s2 = "";
+        String provider = getProvider();
+        if (Strings.isNonBlank(s) && Strings.isNonBlank(provider) && !s.toLowerCase().contains(provider.toLowerCase()))
+            s2 += " "+provider;
+        String region = getRegion();
+        if (Strings.isNonBlank(s) && Strings.isNonBlank(region) && !s.toLowerCase().contains(region.toLowerCase()))
+            s2 += " "+region;
+        String endpoint = getEndpoint();
+        if (Strings.isNonBlank(s) && Strings.isNonBlank(endpoint) && !s.toLowerCase().contains(endpoint.toLowerCase()))
+            s2 += " "+endpoint;
+        s2 = s2.trim();
+        if (Strings.isNonBlank(s)) {
+            if (Strings.isNonBlank(s2)) {
+                return s+" ("+s2+")";
+            }
+            return s;
+        }
+        if (Strings.isNonBlank(s2)) {
```

Comment:
`s2` can only be non-blank if `s` is non-blank and in that case the method has already returned on line 1575.